frontend/test/metabase-visual/collections/bookmarks.cy.spec.js

```diff
@@ -0,0 +1,21 @@
+import { restore, sidebar } from "__support__/e2e/cypress";
+import { getSidebarSectionTitle as getSectionTitle } from "__support__/e2e/helpers/e2e-collection-helpers";
+
+describe("Bookmarks in a collection page", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  it("updates sidebar and bookmark icon color when bookmarking a collection in its page", () => {
+    cy.visit("/collection/1");
+
+    cy.icon("bookmark").click();
+
+    sidebar().within(() => {
+      getSectionTitle("Bookmarks");
+    });
+
+    cy.percySnapshot();
+  });
+});
```

Comment:
Please prefer using the API to set things up unless you're explicitly testing the UI flow. It makes the tests run much faster and it saves us a ton of CI time.

Also, since Percy visual tests cost some money, I think it'd make sense to create a single visual test for the whole collections page. We can do a few things:

* make one of the collection's official to also test the "badge" icon in the sidebar
* bookmark a few entities to cover more UI states (e.g. a question, a model, a dashboard, a collection, an official collection)
* add various entities to a collection that's going to be open during the snapshot and pin a few things to get some more UI states covered 

Everything listed above should be fairly easy to do via API. We can do that in a subsequent PR as well though.

Reply:
@kulyk, thanks.

Bookmark now going via API.

Happy to add more checks in this test.
Soon I'll update the icons we render, and also the official badge, so it makes sense to add checks around those along with their implementation.